lib/phoenix_component.ex

```diff
@@ -290,6 +290,139 @@ defmodule Phoenix.Component do
     quote do
       import Phoenix.LiveView
       import Phoenix.LiveView.Helpers
+      import unquote(__MODULE__), only: [attr: 2, attr: 3]
+
+      Module.register_attribute(__MODULE__, :__attrs__, accumulate: true)
+      Module.register_attribute(__MODULE__, :__components_calls__, accumulate: true)
+      Module.put_attribute(__MODULE__, :__components__, %{})
+
+      @on_definition {unquote(__MODULE__), :__on_definition__}
+      @before_compile {unquote(__MODULE__), :__before_compile__}
```

Comment:
I think those are the default. :)

```suggestion
      @on_definition unquote(__MODULE__)
      @before_compile unquote(__MODULE__)
```